From 18b027f26b4f849cf533bc806368a9ccec3b3975 Mon Sep 17 00:00:00 2001
From: Craig Barnes <craigbarnes@protonmail.com>
Date: Sat, 19 Dec 2020 22:09:03 +0000
Subject: [PATCH] meson: add missing "wl_proto_headers" dependency for pgolib
 and vtlib

Dependency chains:

* pgolib -> terminal.c -> terminal.h -> wayland.h
* vtlib -> csi.c -> config.h -> wayland.h

wayland.h includes <primary-selection-unstable-v1.h>, which must be
generated by a custom_target() rule *before* any sources that require
it are built. Failure to fully specify these dependencies can otherwise
result in a race condition, where a dependent source file gets compiled
(and fails with a "header not found" error) before the header itself
has been generated.
---
 meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/meson.build b/meson.build
index 4370acc..282f736 100644
--- meson.build
+++ meson.build
@@ -122,6 +122,7 @@ vtlib = static_library(
   'osc.c', 'osc.h',
   'sixel.c', 'sixel.h',
   'vt.c', 'vt.h',
+  wl_proto_src + wl_proto_headers,
   version,
   dependencies: [pixman, fcft, tllist],
   link_with: misc,
@@ -132,6 +133,7 @@ pgolib = static_library(
   'grid.c', 'grid.h',
   'selection.c', 'selection.h',
   'terminal.c', 'terminal.h',
+  wl_proto_src + wl_proto_headers,
   dependencies: [pixman, fcft, tllist],
   link_with: vtlib,
 )
